zip-0032.rst

```diff
@@ -309,6 +314,71 @@ Sprout child key derivation
 - Use :math:`I_R` as the child chain code :math:`\mathsf{c}_i`.
 
 
+Specification: Orchard key derivation
+=====================================
+
+The derivation mechanism for Sapling addresses specified above incurs significant complexity to support
+non-hardened derivation. In the several years since Sapling was deployed, we have seen no use cases for
+non-hardened derivation appear. With that in mind, we define Orchard key derivation very similarly to
+Sprout above: only hardened derivation is supported.
+
+Orchard extended keys
+---------------------
+
+We represent an Orchard extended spending key as :math:`(\mathsf{sk, c}),` where :math:`\mathsf{sk}`
+is the normal Orchard spending key (opaque 32 bytes), and :math:`\mathsf{c}` is the chain code.
+
+Orchard master key generation
+-----------------------------
+
+Let :math:`S` be a seed byte sequence of a chosen length, which MUST be at least 32 and at most 252 bytes.
+
+- Calculate :math:`I = \mathsf{BLAKE2b}\text{-}\mathsf{512}(\texttt{“ZcashIP32Orchard”}, S)`.
+- Split :math:`I` into two 32-byte sequences, :math:`I_L` and :math:`I_R`.
+- Use :math:`I_L` as the master spending key :math:`\mathsf{sk}_m`.
+- Use :math:`I_R` as the master chain code :math:`\mathsf{c}_m`.
+- Return :math:`(\mathsf{sk}_m, \mathsf{c}_m)` as the master extended spending key
+  :math:`m_\mathsf{Orchard}`.
+
+Orchard child key derivation
+----------------------------
+
+:math:`\mathsf{CDKsk}((\mathsf{sk}_{par}, \mathsf{c}_{par}), i)`:math:`\rightarrow (\mathsf{sk}_{i}, \mathsf{c}_i)`
+
+- Check whether :math:`i \geq 2^{31}` (whether the child is a hardened key).
+
+  - If so (hardened child): let
+    :math:`I = \mathsf{PRF^{expand}}(\mathsf{c}_{par}, [\texttt{0x81}]\,||\,\mathsf{sk}_{par}\,||\,\mathsf{I2LEOSP}_{32}(i))`.
+  - If not (normal child): return failure.
+
+- Split :math:`I` into two 32-byte sequences, :math:`I_L` and :math:`I_R`.
+- Use :math:`I_L` as the child spending key :math:`\mathsf{sk}_{i}`.
+- Use :math:`I_R` as the child chain code :math:`\mathsf{c}_i`.
```

Comment:
It would be helpful to add "chain code" to the terminology section. @daira explained it as a capability for child key derivation, which makes more sense to me than the BIP 32 definition which specifies it as "additional randomness" but doesn't give a justification for needing that extra entropy.